sdk/signature-mpc-wasm/src/lib.rs

```diff
@@ -219,6 +219,32 @@ pub fn recovery_id_sha256(
     })?.into())
 }
 
+#[wasm_bindgen]
+pub fn generate_keypair() -> JsValue {
+    let (public_key, private_key) = signature_mpc::twopc_mpc_protocols::dwallet_transfer::generate_keypair();
+    serde_wasm_bindgen::to_value(&(public_key, private_key)).unwrap()
+}
+
+#[wasm_bindgen]
+pub fn generate_proof(
+    secret_share: Vec<u8>,
+    public_key: Vec<u8>,
+) -> JsValue {
+    let language_public_parameters = signature_mpc::twopc_mpc_protocols::dwallet_transfer::get_proof_public_parameters(public_key.clone());
+    let proof_public_output =
+        signature_mpc::twopc_mpc_protocols::dwallet_transfer::generate_proof(
+            public_key,
+            secret_share,
+            language_public_parameters,
+        );
+
+    let proof = bcs::to_bytes(&proof_public_output.proof).unwrap();
+    let encypted_discrete_log = bcs::to_bytes(&proof_public_output.encrypted_discrete_log).unwrap();
+    let range_proof_commitment = bcs::to_bytes(&proof_public_output.range_proof_commitment).unwrap();
+
+    serde_wasm_bindgen::to_value(&(proof, encypted_discrete_log, range_proof_commitment)).unwrap()
```

Comment:
Why not serialize the `proof_public_output` directly? That's the whole purpose of this struct to hold it together no?